### PR TITLE
Updating Readme to add a new Clusterrole

### PR DESCRIPTION
**Pull request by ayushsatyam146**
Updating the documentation to include the step
granting RBAC "get", "list", and "watch"
access to the resource being shared to the CSI
Driver's service account via a Cluserrole.
Adding the steps to create the Clusterrole
which will be created by the resource owner.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,28 @@ Sharing resources is done as follows:
...
+       resources: ["configmaps"]
+       resourceNames: ["shared-config"]
+       verbs: ["get", "list", "watch"]
+     - apiGroups: ["sharedresource.openshift.io"]
+       resources: ["sharedconfigmaps", "sharedsecrets"]
+       verbs: ["get", "list", "watch"]
+     - apiGroups: ["authorization.k8s.io"]
+       resources: ["subjectaccessreviews"]
+       verbs: ["create"]
+    ```
```
Comment (apoorvajagtap) [inline README.md:70]: We'll need an additional step to add the clusterrole to the `csi-driver-shared-resource`.
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: openshift-builds-shared-resource-secret-configmap-share-watch-sar-create
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: openshift-builds-shared-resource-secret-configmap-share-watch-sar-create
subjects:
- kind: ServiceAccount
  name: csi-driver-shared-resource
  namespace: openshift-builds
```

**Review by apoorvajagtap — CHANGES_REQUESTED** [review]

**Review by apoorvajagtap — APPROVED** [review]

**Comment by sayan-biswas** [comment]
There's no code change here, and having some issues with konflux ci. 
Hence merging this.

**Pull request merged**